src/gpuarray_buffer_cuda.c

```diff
@@ -1085,13 +1085,21 @@ static inline int gen_extcopy_kernel(const cache_key_t *a,
   int flags = GA_USE_PTX;
   unsigned int bits = sizeof(void *)*8;
   int types[2];
-  const char *in_t;
-  const char *out_t;
+  const char *in_t, *in_ld_t;
+  const char *out_t, *out_ld_t;
   const char *rmod;
   char arch[6]; /* Must be at least 6, see detect_arch() */
 
   in_t = map_t(a->itype);
   out_t = map_t(a->otype);
+  if (a->itype == GA_HALF)
+    in_ld_t = "b16";
```

Comment:
Can you put a comment why this is needed? Why f16 do not work and what b16 mean?